src/compositemark/errorbar.ts

```diff
@@ -107,9 +110,9 @@ export function normalizeErrorBar(
     ...outerSpec,
     transform,
     layer: [
-      ...makeErrorBarPart('ticks', tick, 'lower'),
-      ...makeErrorBarPart('ticks', tick, 'upper'),
-      ...makeErrorBarPart('rule', 'rule', 'lower', 'upper')
+      ...makeErrorBarPart('ticks', tick, 'lower', null, tooltipEncoding),
```

Comment:
I think it's worth making `makeErrorBarPart` take parameters as an object since this object this method is getting a bit hard at this point. 

(I have wanted to change this earlier, but haven't got the chance yet.)

Reply:
Can you clarify this a bit more?

Reply:
See 86eae67bb905be5b6582bf27d2e60f6635918862 for example.  I did the same thing to `tickCount()` there so it will enforce us to include parameter names.

Right now this method is extremely hard to read.  It's hard to remember what the first 'rule' and the second 'rule' are for. By forcing them to be an object, we can understand the parameter name right away.

Just like in Python `f(x=3,y=4,z=5,opt={})` is clearer than `f(3,4,5,{})` as we know that the first parameter is `x`, second is `y` and so on.  For method with 1-2 parameters, this isn't so important. But once we have several, it's becoming quite hard to read.

Reply:
https://basarat.gitbooks.io/typescript/content/docs/tips/functionParameters.html